cmd/bnscli/clitests/README.md

```diff
@@ -0,0 +1,39 @@
+# `bnscli` tool tests
+
+This is a set of tests executing shell scripts and ensuring the output is as
+expected. Tests expect `bnscli` binary to be present in one of the `$PATH`
+directories.
+
+
+### Running tests
+
+To run the tests you need Go. We are using Go's
+[testing](https://golang.org/pkg/testing/) package as the test runner.  Enter
+`clitest` directory and run:
+
+    $ go test .
+
+
+### Adding new test
+
+To add a new test, create a file `<test_name>.test` in this directory. It
+should be a [Bourne shell](https://en.wikipedia.org/wiki/Bourne_shell) (not
+[bash](https://en.wikipedia.org/wiki/Bash_(Unix_shell))) script. Its stdout
+will be captured by the test runner and compared with `<test_name>.test.gold`
+file content.
+
+Best is to start your test file with the following lines:
+
+    #!/bin/sh
+    set -e
+
+
+### Creating a golden file
+
+Do not create `xxx.test.gold` files by hand. Instead, run the test runner with
```

Comment:
:+1:

Reply:
This is an another idea from a great presentation on testing https://www.youtube.com/watch?v=yszygk1cpEc&feature=youtu.be&t=867